pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -0,0 +1,199 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ensurer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	flowcontrolv1beta1 "k8s.io/api/flowcontrol/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta1"
+	flowcontrolapisv1beta1 "k8s.io/kubernetes/pkg/apis/flowcontrol/v1beta1"
+)
+
+var (
+	errObjectNotFlowSchema = errors.New("object is not a FlowSchema type")
+)
+
+// FlowSchemaEnsurer ensures the specified bootstrap configuration objects
+type FlowSchemaEnsurer interface {
+	Ensure([]*flowcontrolv1beta1.FlowSchema) error
+}
+
+// FlowSchemaRemover removes the specified bootstrap configuration objects
+type FlowSchemaRemover interface {
+	Remove([]string) error
+}
+
+// NewSuggestedFlowSchemaEnsurer returns a FlowSchemaEnsurer instance that
+// can be used to ensure a set of suggested FlowSchema configuration objects.
+// shouldCreate indicates whether a missing 'suggested' FlowSchema object should be recreated.
+func NewSuggestedFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface, shouldCreate bool) FlowSchemaEnsurer {
+	wrapper := &flowSchemaWrapper{
+		client: client,
+	}
+	return &fsEnsurer{
+		strategy: newSuggestedEnsureStrategy(wrapper, shouldCreate),
+		wrapper:  wrapper,
+	}
+}
+
+// NewMandatoryFlowSchemaEnsurer returns a FlowSchemaEnsurer instance that
+// can be used to ensure a set of mandatory FlowSchema configuration objects.
+func NewMandatoryFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface) FlowSchemaEnsurer {
+	wrapper := &flowSchemaWrapper{
+		client: client,
+	}
+	return &fsEnsurer{
+		strategy: newMandatoryEnsureStrategy(wrapper),
+		wrapper:  wrapper,
+	}
+}
+
+// NewFlowSchemaRemover returns a FlowSchemaRemover instance that
+// can be used to remove a set of FlowSchema configuration objects.
+func NewFlowSchemaRemover(client flowcontrolclient.FlowSchemaInterface) FlowSchemaRemover {
+	return &fsEnsurer{
+		wrapper: &flowSchemaWrapper{
+			client: client,
+		},
+	}
+}
+
+// GetFlowSchemaRemoveCandidate returns a list of FlowSchema object
+// names that are candidates for deletion from the cluster.
+// bootstrap: a set of hard coded FlowSchema configuration objects
+// kube-apiserver maintains in-memory.
+func GetFlowSchemaRemoveCandidate(client flowcontrolclient.FlowSchemaInterface, bootstrap []*flowcontrolv1beta1.FlowSchema) ([]string, error) {
+	// TODO: use a lister here to avoid periodic LIST calls
```

Comment:
Can you open an issue for it (and switch to TODO(#<issue number>): Use a lister ....

[Same for PL - I mean use a single issue]

Reply:
https://github.com/kubernetes/kubernetes/issues/101667